src/vs/workbench/contrib/chat/browser/chatContentParts/toolInvocationParts/chatTerminalToolProgressPart.ts

```diff
@@ -230,6 +555,14 @@ export class ChatTerminalToolProgressPart extends BaseChatToolInvocationSubPart
 	private _terminalInstance: ITerminalInstance | undefined;
 	private readonly _decoration: TerminalCommandDecoration;
 
+	private readonly _commandStreamingListener: MutableDisposable<IDisposable>;
+	private _streamingCommand: ITerminalCommand | undefined;
+	private _trackedCommandId: string | undefined;
+	private _streamingQueue: IStreamingSnapshotRequest[];
+	private readonly _streamingSnapshotRetryCounts = new WeakMap<ITerminalCommand, number>();
```

Comment:
It's very rare to actually need `WeakMap`, using it makes red flags go off in my mind that the lifecycle of this class is not managed properly.